packages/bsky/tests/views/posts.test.ts

```diff
@@ -0,0 +1,86 @@
+import AtpAgent from '@atproto/api'
+import { TestEnvInfo, runTestEnv } from '@atproto/dev-env'
+import { forSnapshot, processAll, stripViewerFromPost } from '../_util'
+import { SeedClient } from '../seeds/client'
+import basicSeed from '../seeds/basic'
+
+describe('pds posts views', () => {
+  let testEnv: TestEnvInfo
+  let agent: AtpAgent
+  let sc: SeedClient
+
+  beforeAll(async () => {
+    testEnv = await runTestEnv({
+      dbPostgresSchema: 'bsky_views_posts',
+    })
+    agent = new AtpAgent({ service: testEnv.bsky.url })
+    const pdsAgent = new AtpAgent({ service: testEnv.pds.url })
+    sc = new SeedClient(pdsAgent)
+    await basicSeed(sc)
+    await processAll(testEnv)
+  })
+
+  afterAll(async () => {
+    await testEnv.close()
+  })
+
+  it('fetches posts', async () => {
+    const uris = [
+      sc.posts[sc.dids.alice][0].ref.uriStr,
+      sc.posts[sc.dids.alice][1].ref.uriStr,
+      sc.posts[sc.dids.bob][0].ref.uriStr,
+      sc.posts[sc.dids.carol][0].ref.uriStr,
+      sc.posts[sc.dids.dan][1].ref.uriStr,
+      sc.replies[sc.dids.alice][0].ref.uriStr,
+    ]
+    const posts = await agent.api.app.bsky.feed.getPosts(
+      { uris },
+      { headers: sc.getHeaders(sc.dids.alice, true) },
+    )
+
+    expect(posts.data.posts.length).toBe(uris.length)
+    expect(forSnapshot(posts.data.posts)).toMatchSnapshot()
+  })
+
+  it('fetches posts unauthed', async () => {
+    const uris = [
+      sc.posts[sc.dids.alice][0].ref.uriStr,
+      sc.posts[sc.dids.alice][1].ref.uriStr,
+      sc.posts[sc.dids.bob][0].ref.uriStr,
+      sc.posts[sc.dids.carol][0].ref.uriStr,
+      sc.posts[sc.dids.dan][1].ref.uriStr,
+      sc.replies[sc.dids.alice][0].ref.uriStr,
+    ]
+
+    const authed = await agent.api.app.bsky.feed.getPosts(
+      { uris },
+      { headers: sc.getHeaders(sc.dids.alice, true) },
+    )
+    const unauthed = await agent.api.app.bsky.feed.getPosts({
+      uris,
+    })
+    const stripped = authed.data.posts.map((p) => stripViewerFromPost(p))
+    // @TODO uncomment
```

Comment:
I could use an eye on this @devinivy 

I fought with it for like 30 min & feel like I'm missing something obvious

Logging the stripped data looks totally fine, but for some reason in the jest compare, it says that `viewer` is still around 🧐

Reply:
Ah, it was just an issue with `stripViewerFromPost()` not dealing with record-with-media embeds.  Just pushed a fix to this branch 👍

Reply:
well now i feel silly. why tf it was it console logging it out just fine then?

Reply:
ah well 🤷‍♂️

Reply:
Ah, it's all good!  No idea, sounds like a ghost in the machine.  I def spent a little time looking through `unauthed` when the issue was in `stripped`.